modules/doi/includes/utilities.inc

```diff
@@ -900,8 +901,11 @@ function islandora_doi_filter_html($node, $action = NULL) {
       default:
         drupal_set_message(t('@var is not a recognized option for handling HTML in imported DOI metadata.', array('@var' => $action)), 'error');
         break;
-
     }
+
+    // Double encode ampersands regardless of prior filtering
+    $text = str_replace(' & ', ' &amp; ', $text);
+    $text = str_replace(' &amp; ', ' &amp;amp; ', $text);
```

Comment:
@bryjbrown are you sure about the space padding? Could we not have something like `Isl&amp;ora` to encode?

Reply:
@d-r-p Do you have any suggestions for how to do this correctly? I'm struggling to figure out how to get the proper number of encodings for an ampersand (2) regardless of whether the filter is running in strip or encode mode, and regardless of whether or not the ampersand is coming in raw, already encoded, or already double-encoded. I thought this would do it, but its bad.

Reply:
@bryjbrown not sure. but I think that
a) Crossref should _always_ deliver encoded ampersands (otherwise they would break XML standards) - or are there counterexamples to this?
b) Double encoding only makes sense in encoding mode; in strip mode (I guess) users will expect _plain_ text (i.e. single encoded inside XML)
c) Double encoding in encoding mode should only happen on those fields that are treated for HTML
d) The drush script should do _exactly_ the same

Does 
```php
$text = str_replace('&amp;', '&amp;amp;', $text);
```
in encoding mode for the four HTML-fields make sense to you?